### PR TITLE
Hillbrad/new format sri

### PR DESCRIPTION
**Pull request by hillbrad**
Update of Subresource Integrity tests to use new integrity format and crossorigin rules.  Contains some 'stub' tests for algorithm priority that won't pass anywhere yet and will need updating in the future.  No <link> tests yet, just <script>

### CODE REVIEW

File: subresource-integrity/subresource-integrity.html [hunk 1]
```diff
@@ -6,74 +6,176 @@
...
+           e.addEventListener("error", function() {test.done()});
+        }
+        document.body.appendChild(e);
+    };
+
+    new SRIScriptTest(
+        true,
+        "Same-origin script with correct hash.",
+        "matching-digest.js",
+        "sha256-EKclUXH9SRRUv3FmL7bIEV0z2s3EvzHFxzHKCnfHT/E="
```
Comment (devd) [inline subresource-integrity/subresource-integrity.html]: Did we never get around to updating the spec to b64_urlsafe?

File: subresource-integrity/subresource-integrity.html [hunk 2]
```diff
@@ -6,74 +6,176 @@
...
+    new SRIScriptTest(
+        true,
+        "Same-origin script with multiple sha256 hashes, including unknown algorithm.",
+        "matching-digest.js",
+        "sha256-EKclUXH9SRRUv3FmL7bIEV0z2s3EvzHFxzHKCnfHT/E= foo666-deadbeefdeadbeefdeadbeefdeadbeefdeadbeefdead"
+    ).execute();
+
+    // TODO may need to change these to hash-source prefix, e.g. 'sha256' instead of 'SHA-256'
+
+    test(function() {assert_equals(getPrioritizedHashFunction('SHA-512', 'SHA-256'), 'SHA-512')},
```
Comment (devd) [inline subresource-integrity/subresource-integrity.html]: confused.. my reading of the spec doesn't say that getprioritizedhashfunction is an actual function available to pages. What are you calling? I don't see it defined either.

File: subresource-integrity/crossorigin-creds.js.sub.headers [hunk 3]
```diff
@@ -0,0 +1,2 @@
+Access-Control-Allow-Origin: http://{{domains[]}}:{{ports[http][0]}}
```
Comment (devd) [inline subresource-integrity/crossorigin-creds.js.sub.headers]: Maybe we should have a test where ACAO is set to the correct origin but allow creds isn't present and the request is cross-origin=anonymous?

File: subresource-integrity/subresource-integrity.html [hunk 4]
```diff
@@ -6,74 +6,176 @@
...
+        "matching-digest.js",
+        "sha256-EKclUXH9SRRUv3FmL7bIEV0z2s3EvzHFxzHKCnfHT/E=?foo=bar?spam=eggs"
+    ).execute();
+
+    new SRIScriptTest(
+        true,
+        "Same-origin script with unknown algorithm only.",
+        "matching-digest.js",
+        "foo666-foolUXH9SRRUv3FmL7bIEV0z2s3EvzHFxzHKCnfHT/E="
+    ).execute();
```
Comment (devd) [inline subresource-integrity/subresource-integrity.html]: a possibly interesting test could be unknown algorithm and wrong hash (to make sure that unknown algorithm's 'ignore' part doesn't mean known algorithms get ignored)

**Comment by devd** [comment]
accepted, btw.

File: subresource-integrity/subresource-integrity.html (hunk 1 above)
Comment (hillbrad) [inline subresource-integrity/subresource-integrity.html]: Not that I see.

File: subresource-integrity/subresource-integrity.html (hunk 2 above)
Comment (hillbrad) [inline subresource-integrity/subresource-integrity.html]: Noted, will remove these in a future PR.

**Pull request merged**